src/Sparrow/Collections/FastStack.cs

```diff
@@ -51,16 +51,33 @@ public FastStack(int capacity)
 
         public int Count => _size;
 
-        // Removes all Objects from the Stack.
+        /// <summary>
+        /// Removes all objects from the Stack but clearing the backing array unless the type is native.
+        /// </summary>
         public void Clear()
         {
-            Array.Clear(_array, 0, _size);
+            if (typeof(T) == typeof(int) || typeof(T) == typeof(uint) || typeof(T) == typeof(byte) ||
+                typeof(T) == typeof(short) || typeof(T) == typeof(long) || typeof(T) == typeof(ulong) ||
+                typeof(T) == typeof(nint) || typeof(T) == typeof(nuint) || typeof(T) == typeof(IntPtr))
+            {
+                _size = 0;
+                _version++;
```

Comment:
Can we put

```
_size = 0;
_version++;
```

at the beginning of the method so we won't need to repeat it below as well?

Reply:
Yeah.